src/mad_spark_alt/core/provider_router.py

```
@@ -420,58 +598,103 @@ async def extract_document_content(
+                        source_type=MultimodalSourceType.FILE_PATH,
+                        data=str(doc_path_obj.absolute()),
+                        mime_type=mime_type,
+                        file_size=doc_size,
+                    )
+                )
+            else:
+                # Text-based documents: read directly
+                try:
+                    text_content = self._read_text_document(doc_path_obj)
+                    text_contexts.append(f"=== {doc_path_obj.name} ===\n{text_content}")
+                except Exception as e:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)